source/usage-examples/findOne.go

```diff
@@ -0,0 +1,46 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"log"
+)
+
+func main() {
+
+	// Replace the uri string with your MongoDB deployment's connection string.
+	uri := "mongodb+srv://<user>:<password>@<cluster-url>?retryWrites=true&w=majority"
+	ctx := context.TODO()
+	client, err := mongo.Connect(ctx, options.Client().ApplyURI(uri))
+	if err != nil {
+		panic(err)
+	}
+	// Disconnect the client once the function returns.
+	defer func() {
+		if err = client.Disconnect(ctx); err != nil {
+			panic(err)
+		}
+	}()
+	coll := client.Database("sample_mflix").Collection("movies")
+	var result bson.M
```

Comment:
I suggest using `bson.D` here. We want to shy away users from using `bson.M` since it does not guarantee order of keys. It is particularly problematic when using it as a sort document (GODRIVER-1610) or to create / use indexes. @iwysiu added more documentation warnings and safety checks in GODRIVER-1810, but I still think we'd prefer limiting first time user exposure to `bson.M`.

Reply:
Thank you for the note. Going forward, I will avoid using `bson.M` in introductory sections, and will use it sparingly (possible with a warning?) in later sections. I have removed all instances of `bson.M` from the page. Fix: https://github.com/mongodb/docs-golang/pull/2/commits/288eadb7c4ecd3ad4ecb280c604f5efacbfdce3b